Lib/test/test_enum.py

```diff
@@ -864,13 +864,13 @@ class Grades(int, Enum):
             C = 3
             D = 2
             F = 0
-        self.assertEqual(f'{self.Grades.B}', 'B')
+        self.assertEqual(f'{Grades.B}', 'B')
```

Comment:
Looking again, I found that `self.Grades` is defined (using `IntEnum` as base rather than `int, Enum` but I think the result is identical), so the duplicate code to be removed is the Grades declaration in these test methods.

Reply:
> (using `IntEnum` as base rather than `int, Enum` but I think the result is identical)

It is very-identically-ish, because `IntEnum` is defined as 

```python
class IntEnum(int, Enum):
    """
    Enum where members are also (and must be) ints
    """

    def __str__(self):
        return "%s" % (self._name_, )

    def __format__(self, format_spec):
        """
        Returns format using actual value unless __str__ has been overridden.
        """
        str_overridden = type(self).__str__ != IntEnum.__str__
        if str_overridden:
            cls = str
            val = str(self)
        else:
            cls = self._member_type_
            val = self._value_
        return cls.__format__(val, format_spec)
```

But, I think that it is better to keep everything as-is, because the test is named `test_MIXIN_format_warning`. And I think that the initial intent was to test the difference in `def __format__(self, format_spec):` between `int, Enum` and `IntEnum`.

One more thing, this test is even explicitly skipped:

```python
    @unittest.skipIf(
            python_version < (3, 12),
            'mixin-format is still using member.value',
            )
    def test_mixin_format_warning(self):
```

Reply:
OK, then let’s keep the Grades inline definition and the removal of `self`.

> One more thing
We already discussed that — the two tests with the same name are meant for different python version.  The future test is meant as a guard to remember to implement a planned change.  (I have still renamed one of the methods in the patch I sent and that you applied.)

Reply:
This is now removed from `test_enum`.